lib/output.js

```diff
@@ -1146,15 +1146,23 @@ function OutputStream(options) {
     DEFPRINT(AST_Dot, function(self, output){
         var expr = self.expression;
         expr.print(output);
-        if (expr instanceof AST_Number && expr.getValue() >= 0) {
-            if (!/[xa-f.)]/i.test(output.last())) {
-                output.print(".");
+        var prop = self.property;
+        if (RESERVED_WORDS(prop) && output.option("ie8")) {
```

Comment:
Done in https://github.com/mishoo/UglifyJS2/pull/2238/files#diff-00fe37b20ddfb9ecc9c492b0725fa2b2R1150